src/gather.ts

```diff
@@ -272,4 +272,45 @@ export class GatherProvider implements IGatherProvider {
 
     await vscode.commands.executeCommand(Constants.openNotebookCommand, undefined, notebook);
   }
+
+  public async smartSelect(vscCell: vscode.NotebookCell): Promise<void> {
+    vscode.window.withProgress(
+      { location: vscode.ProgressLocation.Notification, title: localize.Common.smartSelecting() }, 
+      async () => {
+        try {
+          const gatheredCode = this.gatherCodeInternal(vscCell);
+          if (gatheredCode.length === 0) {
+            // TODO handle error
```

Comment:
Might be a bit of a big TODO here. Maybe have a bug logged for this if you want to check this in first. Or at least have just a console / trace log so that it get's notified or logged somehow.